src/signature.ts

```diff
@@ -22,79 +21,40 @@ export interface ECDSASignatureBuffer {
  * Returns the ECDSA signature of a message hash.
  */
 export function ecsign(msgHash: Buffer, privateKey: Buffer, chainId?: number): ECDSASignature
-export function ecsign(
-  msgHash: Buffer,
-  privateKey: Buffer,
-  chainId: BN | string | Buffer
-): ECDSASignatureBuffer
+export function ecsign(msgHash: Buffer, privateKey: Buffer, chainId: BNLike): ECDSASignatureBuffer
```

Comment:
This was intentional to split the `BNLike` type to these two overlays and not use the `BNLike` type here. Otherwise there is an ambiguity on the `number` type.

Update: I just checked in the code, I would have expected for `TypeScript` to assume `ECDSASignature | ECDSASignatureBuffer` as a return type when used with a `number` `chainId`, this is not the case though, seems `TypeScript` takes the first working signature.

Will therefore let this pass, feeling a little uneasy on this nevertheless.